src/main/java/core/basesyntax/service/transaction/FruitTransaction.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax.service.transaction;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation getOperationByString(String string) {
+            Map<String, Operation> operationMap = new HashMap<>();
+            operationMap.put("b",BALANCE);
+            operationMap.put("s",SUPPLY);
+            operationMap.put("p",PURCHASE);
+            operationMap.put("r",RETURN);
```

Comment:
Creating and filling the same Map each time when this method is called? Think about implementation.